File: leetcode/0560.go
package leetcode

func subarraySum(nums []int, k int) (ans int) {
	mp := map[int]int{0: 1}
	for i, pre := 0, 0; i < len(nums); i++ {
		pre += nums[i]
		if _, ok := mp[pre-k]; ok {
			ans += mp[pre-k]
		}
		mp[pre]++
	}
	return
}

File: leetcode/0560.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int ans = 0, presum = 0;
        unordered_map<int, int> cnt;
        for (auto&& num : nums) {
            ++cnt[presum];
            presum += num;
            if (cnt.count(presum - k))
                ans += cnt[presum - k];
        }
        return ans;
    }
};